contrib/openssl-cmake/CMakeLists.txt

```diff
@@ -985,8 +981,6 @@ set(CRYPTO_SRC
     ${OPENSSL_SOURCE_DIR}/providers/common/securitycheck.c
     ${OPENSSL_SOURCE_DIR}/providers/common/securitycheck_default.c
     ${OPENSSL_SOURCE_DIR}/providers/defltprov.c
-    ${OPENSSL_SOURCE_DIR}/providers/fips/fips_entry.c
```

Comment:
Hmm, I believe these two files will be needed for OpenSSL's FIPS mode. But I also see that including them here produces duplicate symbols with dynamic linking :-(

What if we remove them here (like you did) and add them conditionally:

```cmake
if(NOT ENABLE_OPENSSL_DYNAMIC)
    set(CRYPTO_SRC ${CRYPTO_SRC}
        ${OPENSSL_SOURCE_DIR}/providers/fips/fips_entry.c
        ${OPENSSL_SOURCE_DIR}/providers/fips/fipsprov.c
    )
endif()
```

Reply:
Done.